packages/MSLG/src/Parser.ts

```diff
@@ -0,0 +1,191 @@
+import * as fs from 'fs-extra';
+import { helpers, ErrorType } from './helpers';
+import * as path from 'path';
+import * as txtfile from 'read-text-file';
+import * as chalk from 'chalk';
+import { MSLGTool } from 'botbuilder-lg';
+
+const readlineSync = require('readline-sync');
+
+export class Parser {
+    private tool: MSLGTool = new MSLGTool();
+
+    public async Parser(program: any) {
+        let filesToParse: any[] = [];
+        let folderStat: any;
+        if (program.in) {
+            filesToParse.push(program.in);
+        }
+
+        if (program.lg_folder) {
+            try {
+                folderStat = fs.statSync(program.lg_folder);
+            } catch (err) {
+                process.stderr.write(chalk.default.redBright('Sorry, ' + program.lg_folder + ' is not a folder or does not exist'));
+                return;
+            }
+            if (!folderStat.isDirectory()) {
+                process.stderr.write(chalk.default.redBright('Sorry, ' + program.lg_folder + ' is not a folder or does not exist'));
+                return;
+            }
+            if (program.subfolder) {
+                filesToParse = helpers.findLGFiles(program.lg_folder, true);
+            } else {
+                filesToParse = helpers.findLGFiles(program.lg_folder, false);
+            }
+            if (filesToParse.length === 0) {
+                process.stderr.write(chalk.default.redBright('Sorry, no .lg files found in the specified folder.'));
+                return;
+            }
+        }
+
+        if ((filesToParse === undefined || filesToParse.length === 0) && program.stdin === undefined) {
+            process.stderr.write(chalk.default.redBright(`Sorry, no .lg files are provided.` + '\n'));
+            return;
+        }
+
+        let errors: string[] = [];
+
+        while (filesToParse.length > 0) {
+            let file = filesToParse[0];
+            try {
+                const parseRes = await this.parseFile(file, program.verbose);
+                if (parseRes === undefined) {
+                    return;
+                }
+
+                errors = errors.concat(parseRes);
+            } catch (err) {
+                process.stderr.write(err);
```

Comment:
Its good practice to have this just throw and have the main caller handle it.

Reply:
fixed. Throw all exceptions in parser.ts and expander.ts to called and print them in the same format.